include/deal.II/base/numbers.h

```diff
@@ -621,21 +621,33 @@ namespace numbers
 } // namespace numbers
 
 
-// Forward declarations
+// Forward declarations.  These are all declared in
+// differentiation/ad/ad_number_traits.h and
+// differentiation/ad/sacado_number_traits.h
 namespace Differentiation
 {
   namespace AD
   {
     namespace internal
     {
-      // Defined in differentiation/ad/ad_number_traits.h
       template <typename T>
       struct NumberType;
     } // namespace internal
 
     // Defined in differentiation/ad/ad_number_traits.h
     template <typename NumberType>
     struct is_ad_number;
```

Comment:
It is odd that we have this here but not the Sacado versions.

Reply:
Yes.